apps/questdk/src/tokens/erc1155.ts

```diff
@@ -15,15 +15,37 @@ export async function fetchERC1155Metadata(
   contractAddress: string,
   tokenId: number,
 ): Promise<any> {
-  const tokenURI: string = await (client.readContract({
+  let tokenURI: string = await (client.readContract({
     address: contractAddress as Address,
     abi: [
-      'function uri(uint256 tokenId) external view returns (string memory)',
+      {
+        inputs: [
+          {
+            internalType: 'uint256',
+            name: 'tokenId',
+            type: 'uint256',
+          },
+        ],
+        name: 'uri',
+        outputs: [
+          {
+            internalType: 'string',
+            name: '',
+            type: 'string',
+          },
+        ],
+        stateMutability: 'view',
+        type: 'function',
+      },
     ],
     functionName: 'uri',
-    args: [tokenId],
+    args: [BigInt(tokenId)],
   }) as Promise<string>)
 
+  if (tokenURI.startsWith('ipfs://')) {
+    const ipfsIdentifier = tokenURI.split('/')[2]
```

Comment:
Similarly to the other repo, we may want to grab the entire path (index 2 and beyond!)

Reply:
good call! fixed